### PR TITLE
docs: troubleshooting webview loading error (rhdevdocs-4803)

### PR DESCRIPTION
**Pull request by deerskindoll**
## What does this pull request change?
new troubleshooting page about webview loading error in incognito/private mode

## What issues does this pull request fix or reference?
https://issues.redhat.com/browse/RHDEVDOCS-4803

## Specify the version of the product this pull request applies to
7.60

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/main/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by azatsarynnyy — APPROVED** [review]
LGTM
Thanks, @deerskindoll for updating the `Troubleshooting` chapter.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+:_content-type: PROCEDURE
+:description: Troubleshooting Webview loading error
+:keywords: user-guide, troubleshooting-webview-loading-error
+:navtitle: Troubleshooting Webview loading error
+:page-aliases: .:troubleshooting-webview-loading-error.adoc
+
+[id="troubleshooting-webview-loading-error"]
+
+= Troubleshooting Webview loading error
```
Comment (max-cx) [inline modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc:9]: Looks like `webview` is lower-case in sentences and sentence capitalization:
https://code.visualstudio.com/api/ux-guidelines/webviews
https://code.visualstudio.com/api/extension-guides/webview

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+pass:[<!-- vale RedHat.Spelling = NO -->]
+
+If you use Microsoft Visual Studio Code - Open Source in an incognito or private browser window, you can encounter an issue with Webviews. The Webview is not loaded and an error message appears: *Error loading webview: Error: Could not register service workers*.
```
Comment (max-cx) [inline modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: If you want, you can shorten it even further and not have to repeat the same info in text and in the quoted error message (which seems to be good enough and self-explanatory):
```suggestion
If you use Microsoft Visual Studio Code - Open Source in a private browsing window, you might encounter the following error message: *Error loading webview: Error: Could not register service workers*.
```

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+The issue affects the following browsers:
+
+* Google Chrome in Incognito mode
+* Mozilla Firefox in Private Browsing mode
```
Comment (max-cx) [inline modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: BTW, there are private browsing modes in other browsers:
* https://blogs.opera.com/news/2014/10/how-to-open-private-window-opera-for-computers/
* https://support.microsoft.com/en-us/microsoft-edge/browse-inprivate-in-microsoft-edge-e6f47704-340c-7d4f-b00d-d0cf35aa1fcc

If the error might appear in any other browser, then it might be worth to consider removing this list if the SME agrees. What does the SME think?

```suggestion
```

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+[WARNING]
```
Comment (max-cx) [inline modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: It find it a bit confusing: the warning is about Mozilla and the procedure is about Chrome.
Let me think if I can suggest some alternative.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+.Fixing Webview loading error in Google Chrome Incognito mode
+
+Change the cookie settings in Google Chrome to fix the issue.
+
+.Procedure
+
+. Open *Settings*.
+. Select *Privacy and security*.
+. Click *Cookies and other site data*.
+. Select *Allow all cookies*.
```
Comment (max-cx) [inline modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: I assume the reader still has the option of not using a private window, so it'd be worth letting the reader know that not using a private window solves this problem too. Maybe something to confirm with the SME.

Another way of presenting all of this information is to turn it into a table like this, but this obviously calls for an SME's review:

```suggestion
.Dealing with the webview error in a private browsing window

[cols="25h,~"]
|===
| Browser | Workarounds

| Google Chrome
|
* Go to *Settings* -> *Privacy and security* -> *Cookies and other site data* -> *Allow all cookies*.

* Do not use an Incognito window.

| Mozilla Firefox
|
* Do not use a private window.

* See link:https://bugzilla.mozilla.org/show_bug.cgi?id=1320796[Bug 1320796: Support ServiceWorkers in Private Browsing Mode].

|===

```

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 3 above)
Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: We just faced the problem in these two browsers. We didn't check it for other browsers.
That's the reason why the specific browsers are mentioned.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 4 above)
Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: just to clarify: the idea is that there's no workaround for Firefox, but it exists for Chrome.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 5 above)
Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: > I assume the reader still has the option of not using a private window, so it'd be worth letting the reader know that not using a private window solves this problem too. Maybe something to confirm with the SME.

That's correct. As it's mentioned at the beginning of this doc, the issue is relevant for private/incognito modes only.

Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: I like the idea of presenting workarounds as a table :+1:

**Comment by azatsarynnyy** [comment]
Thanks, @max-cx for reviewing the content. I addressed some of your comments.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 3 above)
Comment (max-cx) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: I guess stating specific browsers can be interpreted that other browsers that are not mentioned (have been tested and) are not affected. That's why I suggested removing this list.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 4 above)
Comment (max-cx) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: Yes, yes. I got that. Thank you.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 3 above)
Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: I see your point @max-cx. Thanks for the clarification.
I agree - let's remove the browsers' names.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 1 above)
Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc:9]: +1

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 2 above)
Comment (azatsarynnyy) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: agree, it sounds better with your suggestion @max-cx 
Thank you!

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 1 above)
Comment (deerskindoll) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc:9]: it is but Vale says otherwise atm. I'll have to update the error rules first.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 5 above)
Comment (deerskindoll) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: I see your point but I think not using a private window in Firefox isn't really a workaround. That's why I used a warning instead. 
I implemented your suggestion to get rid of the list but I think I should keep it, actually. Maybe rephrase it a little bit to emphasize we don't know if other browsers have the same issue. Because a table with workarounds implies the same as a list, doesn't it? That we have tested the browsers and we have solutions for the error for each browser.

Comment (deerskindoll) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: Update: the incognito mode does more than just block cookies so even if you enable cookies, other things such as site data still won't be saved.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc (hunk 3 above)
Comment (deerskindoll) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc]: I updated the wording, please let me know if it's ok with you now.

**Review by max-cx — APPROVED** [review]
Pre-approving if I am not available when you need this approval.

File: modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc [hunk 6]
```diff
@@ -0,0 +1,38 @@
+:_content-type: PROCEDURE
+:description: Troubleshooting webview loading error
+:keywords: user-guide, troubleshooting-webview-loading-error
+:navtitle: Troubleshooting webview loading error
+:page-aliases: .:troubleshooting-webview-loading-error.adoc
+
+[id="troubleshooting-webview-loading-error"]
+
+pass:[<!-- vale RedHat.Spelling = NO -->]
```
Comment (max-cx) [inline modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc:11]: These two lines, 8 and 9, are probably causing the rendering issue.

Comment (max-cx) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc:11]: Can you move `pass:[<!-- vale RedHat.Spelling = NO -->]` to a line after 11?

Comment (max-cx) [reply modules/end-user-guide/pages/troubleshooting-webview-loading-error.adoc:11]: And maybe try to have no empty lines between the id line and the title line?
